nextcord/application_command.py

```diff
@@ -1240,76 +1262,180 @@ def __init__(
+                ]
+                literals: List[Annotated[OptionConverter, object]] = []
+            else:
+                unpacked_annotations, literals = unpack_annotation(
+                    parameter.annotation, list(self.option_types.keys())
+                )
+            # Make sure that all literals are only OptionConverters and nothing else.
+            for lit in literals:
+                if not isinstance(lit, OptionConverter) or lit is not None:
+                    raise ValueError(
+                        f"{self.error_name} You cannot use non-OptionConverter literals when the base annotation is "
+                        f"not Literal."
+                    )
+
+            # Pyright gets upset at appending these lists together, but not upset when .extend is used?
+            # grouped_annotations: List[Union[type, Annotated[object, OptionConverter]]] = unpacked_annotations + \
+            #                                                                              literals
+            grouped_annotations: List[
+                Union[type, Annotated[Optional[OptionConverter], object], Type[None]]
+            ] = []
+            grouped_annotations.extend(unpacked_annotations)
+            grouped_annotations.extend(literals)
+            # The only literals in this should be OptionConverters. Anything else should have triggered the ValueError.
+            for anno in grouped_annotations:
+                if isinstance(anno, object) and isinstance(anno, OptionConverter):
+                    # If the annotation is instantiated, add it to the converters and set the anno to the type it has.
+                    annotation_converters.append(anno)
+                    anno = anno.type
+                elif isinstance(anno, type) and issubclass(anno, OptionConverter):
+                    # If the annotation is NOT instantiated, instantiate it and do the above.
+                    made_converter = anno()
+                    annotation_converters.append(made_converter)
+                    anno = made_converter.type
+
+                if anno is None or anno == type(None):  # type: ignore
+                    # Ignore whatever your IDE tells you about the `==`, using `is` makes it not work.
+                    # If None is included, they want it to be optional. But we don't want None processed fully as anno.
+                    annotation_required = False
+                else:
+                    if found_type is MISSING:
+                        # If we haven't set the type of the annotation, set it.
+                        found_type = self.get_type(anno)
+                    elif self.get_type(anno) is not found_type:
+                        raise ValueError(
+                            f"{self.error_name} | Annotation {anno} is incompatible with {found_type}"
+                        )
+
+                    # Annotation specific changes.
+                    if anno is CategoryChannel:
```

Comment:
I feel like what could be done instead is a mapping from the type of channel to its corresponding enum
Then you grab from this mapping via the type of `anno`

Reply:
Suggested by ooliver as well, finally implemented.